media/jsonb/server/src/main/java/io/helidon/media/jsonb/server/JsonBindingSupport.java

```diff
@@ -0,0 +1,117 @@
+                                                ? super ServerResponse,
+                                                ? extends Jsonb> jsonbProvider) {
+        super();
+        this.jsonbProvider = Objects.requireNonNull(jsonbProvider);
+    }
+
+    @Override
+    public void update(final Routing.Rules routingRules) {
+        routingRules.any(this);
+    }
+
+    @Override
+    public void accept(final ServerRequest request, final ServerResponse response) {
+        final Jsonb jsonb = this.jsonbProvider.apply(request, response);
+        request.content()
+            .registerReader(cls -> true,
+                            JsonBinding.reader(jsonb));
+        response.registerWriter(payload -> wantsJson(request, response),
+                                JsonBinding.writer(jsonb));
+        request.next();
+    }
+
+    private static boolean wantsJson(final ServerRequest request, final ServerResponse response) {
+        final boolean returnValue;
+        final MediaType outgoingMediaType = response.headers().contentType().orElse(null);
+        if (outgoingMediaType == null) {
+            final MediaType preferredType;
+            final Collection<? extends MediaType> acceptedTypes = request.headers().acceptedTypes();
+            if (acceptedTypes == null || acceptedTypes.isEmpty()) {
+                preferredType = MediaType.APPLICATION_JSON;
+            } else {
+                preferredType = acceptedTypes
+                    .stream()
+                    .map(type -> {
+                            if (type.test(MediaType.APPLICATION_JSON)) {
+                                return MediaType.APPLICATION_JSON;
+                            } else if (type.hasSuffix("json")) {
+                                return MediaType.create(type.type(), type.subtype());
+                            } else {
+                                return null;
+                            }
+                        })
+                    .filter(Objects::nonNull)
+                    .findFirst()
+                    .orElse(null);
+            }
+            if (preferredType == null) {
+                returnValue = false;
+            } else {
+                response.headers().contentType(preferredType);
+                returnValue = true;
+            }
+        } else {
+            returnValue = MediaType.JSON_PREDICATE.test(outgoingMediaType);
+        }
+        return returnValue;
+    }
+
+    /**
+     * Creates a new {@link JsonBindingSupport}.
+     *
+     * @param jsonbProvider a {@link BiFunction} that returns a {@link
+     * Jsonb} when given a {@link ServerRequest} and a {@link
+     * ServerResponse}; must not be {@code null}
+     *
+     * @return a new {@link JsonBindingSupport}
+     *
+     * @exception NullPointerException if {@code jsonbProvider} is
+     * {@code null}
+     */
+    public static JsonBindingSupport create(final BiFunction<? super ServerRequest,
```

Comment:
Is this really needed as the only method? If there is a possibity to use a single Jsonb instance created as a default instance, please add also method create() with no parameters.

Reply:
Also why is this request/response dependent? I do not know JSON-B (or Jackson), so please just explain why it makes sense to do this, rather than just providing an instance.

Reply:
I'll add `create()` as well.  Normally I don't like `static` instances but it sounds like that's OK so I'll follow the JSON-P support conventions here.

Regarding request/response-based instance creation, often non-trivial applications using Jackson (I don't use Yasson, but I assume it's similar) need different mapping setups for differing JAX-RS resource classes—it is common to create an `ObjectMapper` for certain collections of resources.  Here, the analogous thing to do would be to allow the caller to decide whether a new `Jsonb` is called for (or a cached one) for a particular request/response pair.